### PR TITLE
New copy fields, clean up user_capabilities logic

### PR DESCRIPTION
**Pull request by AlanCoding**
Connect https://github.com/ansible/awx/issues/1167

I still need to look at tests, and I may not be completely finished with this yet, but it seems to be functional as I go through the table of resources that need the new copy capability.

Some of this is me trying to pay down technical debt that I previously incurred. I really wanted to implement this efficiently, but I was not very good at it before. I put `capabilities_prefetch` on the view, whereas the prefetch list was on the serializer. This was very hard to follow. Possibly even worse, I forced the paginated queryset to execute prematurely by adding a gory special condition in `generics.py`.

Some sacrifices to the gods of polymorphism are still needed, but this is now completely serializer-based, and I am using a trick I learned with the serializer context. This hat trick was first employed here:

https://github.com/ansible/awx/blob/c35c01e7b143d7d44062643d5f57c5854ce6e661/awx/api/serializers.py#L4172-L4180

The goal here is kind of a "prefetch over a page with this custom logic" type thing.

Note: ultimately I will see about migrating the organization counts to this mechanism as well. It's the same general kind of situation. All 3 involve the same general category of "hack", and leveraging the context is the most elegant form I have for it.

### CODE REVIEW

**Comment by AlanCoding** [comment]
Sorry that WIP label took a while, I was experiencing some of the common headaches with mocking things inside the access code (I think there's some bad behavior somewhere that is messing up the logical behavior). This one tricky part really needed coverage, and I think I have something which is at least passable at this point.

The database queries look pretty good.

We need to optimize the unified job list. I leave this as an exercise for later.

File: awx/api/serializers.py [hunk 1]
```diff
@@ -1146,7 +1197,11 @@ def validate_variables(self, value):
 
 
 class InventorySerializer(BaseSerializerWithVariables):
-    show_capabilities = ['edit', 'delete', 'adhoc']
+    show_capabilities = ['edit', 'delete', 'adhoc', 'copy']
```
Comment (AlanCoding) [inline awx/api/serializers.py:1511]: @jakemcdermott here you can see where the lists grew by one new field, the "copy" field. Hopefully this should cover everything in the spec (but alas, still a PR).

The rest of this is really brutal stuff with using @wwitzel3's new roles to prefetch the fields for list views.

File: awx/main/models/unified_jobs.py [hunk 2]
```diff
@@ -180,12 +180,6 @@ def unique_error_message(self, model_class, unique_check):
         else:
             return super(UnifiedJobTemplate, self).unique_error_message(model_class, unique_check)
 
-    @classmethod
-    def invalid_user_capabilities_prefetch_models(cls):
-        if cls != UnifiedJobTemplate:
-            return []
-        return ['project', 'inventorysource', 'systemjobtemplate']
-
```
Comment (AlanCoding) [inline awx/main/models/unified_jobs.py:188]: I genuinely feel bad that I ever introduced this method in the first place.

File: awx/api/generics.py [hunk 3]
```diff
@@ -355,13 +355,6 @@ class ListAPIView(generics.ListAPIView, GenericAPIView):
     def get_queryset(self):
         return self.request.user.get_queryset(self.model)
 
-    def paginate_queryset(self, queryset):
```
Comment (matburt) [inline awx/api/generics.py:358]: hah, dead code

File: awx/main/models/unified_jobs.py (hunk 2 above)
Comment (matburt) [reply awx/main/models/unified_jobs.py:188]: Atonement accepted, Your sins are forgiven

**Review by matburt — APPROVED** [review]

**Review by wwitzel3 — APPROVED** [review]
This looks great!
